src/jvmMain/kotlin/io/beatmaps/api/collaboration.kt

```diff
@@ -81,17 +85,59 @@ fun Route.collaborationRoute() {
             val req = call.receive<CollaborationResponseData>()
 
             val success = transaction {
-                if (req.accepted) {
-                    Collaboration.update({
-                        Collaboration.id eq req.collaborationId and (Collaboration.collaboratorId eq sess.userId)
-                    }) {
-                        it[accepted] = true
-                    } > 0
-                } else {
-                    Collaboration.deleteWhere {
-                        Collaboration.id eq req.collaborationId and (Collaboration.collaboratorId eq sess.userId)
-                    } > 0
+                val mapId =
+                    if (req.accepted) {
+                        Collaboration.updateReturning(
+                            {
+                                Collaboration.id eq req.collaborationId and (Collaboration.collaboratorId eq sess.userId)
+                            },
+                            {
+                                it[accepted] = true
+                            },
+                            Collaboration.mapId
+                        )
+                    } else {
+                        Collaboration.deleteWhere {
+                            Collaboration.id eq req.collaborationId and (Collaboration.collaboratorId eq sess.userId)
+                        }
+
+                        null
```

Comment:
Does this not mean that success is now always false for deletes (as mapId = null)